packages/binding-modbus/src/modbus-client-factory.ts

```diff
@@ -19,12 +19,26 @@ const debug = createDebugLogger("binding-modbus", "modbus-client-factory");
 
 export default class ModbusClientFactory implements ProtocolClientFactory {
     public readonly scheme: string = "modbus+tcp";
+    private singleton: ModbusClient;
 
     public getClient(): ProtocolClient {
-        debug(`Creating client for '${this.scheme}'`);
-        return new ModbusClient();
+        debug(`Get client for '${this.scheme}'`);
+        if (!this.singleton) {
+            this.init();
+        }
+        return this.singleton;
     }
 
-    public init = (): boolean => true;
-    public destroy = (): boolean => true;
+    public init(): boolean {
```

Comment:
Does this method need to be `public`?

I assume it is `public` to allow recreating the singleton instance. IF so would choose a better name...

Reply:
SHOULD it be possible to recreate the singleton from the outside?

Reply:
I think is coming from [ClientFactoryInterface](https://github.com/eclipse/thingweb.node-wot/blob/3acb1aab2d0c1a277896c8a871a2a92910ee2554/packages/core/src/protocol-interfaces.ts#L79) and how its lifecycle is treated in the core. I don't think I can make it private/protected.

Reply:
I see and I agree.

But the current implementation, whenever `init()` is called, it re-creates the singleton.
I wonder whether it makes sense to add a check like the following since it can be triggered/called from the outside

```
public init(): boolean {
	if (!this.singleton) {
		debug(`Initializing client for '${this.scheme}'`);
        	this.singleton = new ModbusClient();
	}
        return true; // not sure if it should report false in case the singleton was already created.. I don't think so
}
```

Having said that, I don't know how likely (or if all) it is possible that init() is called multiple times.

Anyhow, "singleton" to me means that the class itself needs to take care that there is just one instance, which cannot change over time....
Do you see my point?

Reply:
I see, good point! pushing fix soon